feat/webpage

gentlementlegen opened the pull request:
Resolves #102
Depends on https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/170

---

## What's new
<img width="983" alt="image" src="https://github.com/user-attachments/assets/41e286a8-2c12-4ef3-ae93-1446905829f9">

The changes aim to help developers running the plugin locally and have a preview of the results that displays very similarly to GitHub. They also benefit from caching, so subsequent runs won't refetch all the data allowing for fast runs. OpenAI also can be spoofed to avoid consuming API credits.

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

gentlementlegen commented:
@whilefoo if you wanna have a look as well. Needs some more tweaks probably but it is reviewable.

gentlementlegen commented on src/helpers/get-comment-details.ts:
```diff
@@ -3,6 +3,7 @@ import { GitHubIssueComment } from "../github-types";
 import { ContextPlugin } from "../types/plugin-input";
 import { QUERY_COMMENT_DETAILS } from "../types/requests";
 
+// TODO: fix batch of 100 nodes
```
Don't know if that should be taken care of in a separate PR.

gentlementlegen commented:
@whilefoo @Keyrxng if you can take a look! I'll fix the CI runs very soon.

Keyrxng commented:
Sorry @gentlementlegen if there's any confusion but I've been told by Rndquu not to perform review until things are cleared up regarding my status, I thought 0x4007 would have mentioned it to you sorry man.

rndquu commented:
@gentlementlegen 

Running `bun run ui:dev` and getting the `POST http://localhost:3000/ net::ERR_CONNECTION_REFUSED` error:
<img width="1511" alt="Screenshot 2024-11-18 at 18 29 50" src="https://github.com/user-attachments/assets/64901de1-8c58-4c39-8b26-81f3e4925bf5">

What am I doing wrong and why do we need to run some service at `http://localhost:3000` ?

gentlementlegen commented:
You are running the command that is used to dev the UI only. We need a server because it runs the plug-in. You should use the root "bon start server" command after compiling the front-end.

rndquu commented:
> You are running the command that is used to dev the UI only. We need a server because it runs the plug-in. You should use the root "bun start server" command after compiling the front-end.

So I need to:
1. Run `bun ui:build` from the `src/web` folder
2. Run `bun server` from the root folder

Correct?

rndquu commented:
@gentlementlegen 

Getting 500 error on "Generate" click:
```
$ bun --watch src/web/api/index.ts
Started server http://localhost:3000
{
  type: 45,
  schema: {
    default: "info",
    [Symbol(TypeBox.Hint)]: "Enum",
    [Symbol(TypeBox.Kind)]: "Union",
    anyOf: [
      [Object ...], [Object ...], [Object ...], [Object ...], [Object ...]
    ],
  },
  path: "/logLevel",
  value: undefined,
  message: "Expected required property",
  errors: [],
}
1 | /** The base Error type thrown for all TypeBox exceptions  */
2 | export class TypeBoxError extends Error {
3 |     constructor(message) {
4 |         super(message);
            ^
error: Unable to decode value as it does not match the expected schema
```

<img width="1511" alt="Screenshot 2024-11-18 at 18 40 45" src="https://github.com/user-attachments/assets/8b278af1-a51a-43e3-a69f-8d710edc3368">

What am I doing wrong?

gentlementlegen commented on src/web/.ubiquity-os.config.yml:
```diff
@@ -0,0 +1,12 @@
+# This file serves as a base to elaborate a custom configuration
```
Tu build would be better? It's just a baseline of a functional config will everything on default.

0x4007 replied:
Create is a better word.

gentlementlegen commented:
@rndquu I removed it from the configuration as it should come from the environment, let me know if that solved your problem. Otherwise maybe something in your env is wrong?

rndquu commented:
> @rndquu I removed it from the configuration as it should come from the environment, let me know if that solved your problem. Otherwise maybe something in your env is wrong?

Getting another 500 error:
```
$ bun --watch src/web/api/index.ts
Started server http://localhost:3000
{
  type: 45,
  schema: {
    [Symbol(TypeBox.Kind)]: "String",
    type: "string",
  },
  path: "/evmPrivateEncrypted",
  value: undefined,
  message: "Expected required property",
  errors: [],
}
1 | /** The base Error type thrown for all TypeBox exceptions  */
2 | export class TypeBoxError extends Error {
3 |     constructor(message) {
4 |         super(message);
            ^
error: Unable to decode value as it does not match the expected schema
```

Don't fully understand where `evmPrivateEncrypted` should be placed.

Also notice that the [latest](https://github.com/ubiquity-os-marketplace/text-conversation-rewards/actions/runs/11903102706) jest run is failing.

gentlementlegen commented:
@rndquu The evm private key should come from the env. But I also added a check to pick it up from the configuration itself [here](https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/181/commits/c1c727faf8fdf7e32b9b4596038e2e4c719693bc). Updated the README as well.

gentlementlegen commented:
@rndquu I will add the one we use for tests in the `.dev.example` then since it would be safe to use.

Now that we have the SDK, every plugin can run as a "server". This server is able to listen to network events. So I think what you describe can be solved by using `.http` files (or curl, or postman or any tool capable of doing network calls) that can have a payload that mimics GitHub one. We used this technique inside other plugins, like here: https://github.com/ubiquity-os-marketplace/command-query/blob/cd27f14a67b1eb090bc7f1e3ef597479f67640e3/tests/http/request.http.
